Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/NamespaceSynchronizationConfig.swift

```diff
@@ -0,0 +1,184 @@
+import Foundation
+import MongoSwift
+import MongoMobile
+
+/**
+ The synchronization portal for this namespace.
+
+ Namespace configurations contain a set of document configurations.
+
+ Configurations are stored both persistently and in memory, and should
+ always be in sync.
+ */
+internal struct NamespaceSynchronization: Sequence {
+    /// The actual configuration to be persisted for this namespace.
+    class Config: Codable, Hashable {
+        fileprivate enum CodingKeys: CodingKey {
+            case syncedDocuments, namespace
+        }
+        /// the namespace for this config
+        let namespace: MongoNamespace
+        /// a map of documents synchronized on this namespace, keyed on their documentIds
+        fileprivate var syncedDocuments: [HashableBSONValue: CoreDocumentSynchronization.Config]
+
+        init(namespace: MongoNamespace,
+             syncedDocuments: [HashableBSONValue: CoreDocumentSynchronization.Config]) {
+            self.namespace = namespace
+            self.syncedDocuments = syncedDocuments
+        }
+        static func == (lhs: NamespaceSynchronization.Config, rhs: NamespaceSynchronization.Config) -> Bool {
+            return true
+        }
+
+        func hash(into hasher: inout Hasher) {
+            namespace.hash(into: &hasher)
+            syncedDocuments.hash(into: &hasher)
+        }
+    }
+
+    typealias Element = CoreDocumentSynchronization
+    typealias Iterator = NamespaceSynchronizationIterator
+
+    /// Allows for the iteration of the document configs contained in this instance.
+    struct NamespaceSynchronizationIterator: IteratorProtocol {
+        typealias Element = CoreDocumentSynchronization
```

Comment:
[q] not redundant with wider-scoped typealias above?

Reply:
when you implement a Swift protocol with associated types (the half-baked Swift feature that lets you do generic interfaces), it's generally idiomatic to use typealias to specify the type parameter of the protocol you're implementing

Reply:
kk, so even though it's the same typealias as above, this has some semantic meaning? or is it purely a style thing?